Getting started/making your first request/update (#237)
vtexdocs/dev-portal-content: docs/guides/Getting-Started/getting-started/making-your-first-request.md

---
title: "Making your first request"
slug: "making-your-first-request"
excerpt: "Learn how to make an API request"
hidden: false
createdAt: "2020-01-15T18:58:34.632Z"
updatedAt: "2021-03-26T14:04:17.887Z"
---

You can use the `TEST METHOD` button in [VTEX API reference](https://developers.vtex.com/docs/api-reference) to test API requests as you read the documentaion. In this tutorial you will learn how to use this feature.

## Before starting

> Each request to VTEX APIs [must be authenticated](https://developers.vtex.com/docs/guides/getting-started-authentication). And to use the Developers Portal `TEST METHOD`, you must use [application keys](https://developers.vtex.com/docs/guides/getting-started-authentication#application-keys).

Reviewer: ```suggestion
Every request to VTEX APIs [must be authenticated](https://developers.vtex.com/docs/guides/getting-started-authentication), either through application keys or user tokens. In the Developers Portal, the `TEST METHOD` requires a pair of an `appKey` and an `appToken`.
```

Author: Boa! Implementei com adaptações

> So before going forward with this tutorial, make sure you have a app key and token pair with the right [permissions](https://help.vtex.com/en/tutorial/roles--7HKK5Uau2H6wxE1rH5oRbc). If you do not have an app key and token pair, learn more about how to [create application keys](https://help.vtex.com/en/tutorial/application-keys--2iffYzlvvz4BDMr6WGUtet).

Reviewer: ```suggestion
Hence, before proceeding with this guide, make sure you have a valid pair of `appKey` and appToken` with the necessary [permissions](https://help.vtex.com/en/tutorial/roles--7HKK5Uau2H6wxE1rH5oRbc) to execute the requests you want. If you do not have this credential yet, learn more about how to [create application keys](https://help.vtex.com/en/tutorial/application-keys--2iffYzlvvz4BDMr6WGUtet).
```

## Trying out a simple GET

>⚠️ The example below is based on the [List orders endpoint](https://developers.vtex.com/docs/api-reference/orders-api#get-/api/oms/pvt/orders), but these instructions apply for any endpoint you wish to try. Feel free to follow along.

To try out an API request follow these steps:

1. Head to the API reference page of the endpoint you wish to test, such as the [List orders endpoint](https://developers.vtex.com/docs/api-reference/orders-api#get-/api/oms/pvt/orders).

> ![List orders page](./making-your-first-request-1)

Reviewer: Faltou indicar os formatos das imagens

8. Fill in the environment. If you do not know which environment to use, try `vtexcommercebeta`.

![application keys and base url fields](./making-your-first-request-2)

9. Click the `TEST METHOD` button. This will send your request to the desired API endpoint.
10. Check your response below the request.

![request and response](./making-your-first-request-3)

If you get a `401 - Unauthorized` as response, review the [application key and token](https://developers.vtex.com/docs/guides/getting-started-authentication#application-keys) and try again. Make sure that the right [permissions](https://help.vtex.com/en/tutorial/application-keys--2iffYzlvvz4BDMr6WGUtet) are defined in the [roles](https://help.vtex.com/en/tutorial/roles--7HKK5Uau2H6wxE1rH5oRbc) and that you copied the exact key and token values into the headers of the request.

>ℹ️ Learn more about [VTEX authentication for developers](https://developers.vtex.com/docs/guides/getting-started-authentication).

If you received a `200 OK` response, it means you got your a successful request to VTEX APIs.

> >ℹ️ Some API requests may return other status codes in case of success, such as `201 Created` or `204 No Content`.

Reviewer: Talvez essa parte possa vir em uma outra seção (e.g., handling http status codes) no final da doc e mencionada aqui

Author: Pode ser, mas a princípio enxergo como parte do passo-a-passo. 🤔

> ## Exporting API information

Reviewer: ```suggestion
## Using an API testing platform
```

> You can export API specification or example requests to be used with [Postman](#using-postman) or the API platform of your choice.

Reviewer: ```suggestion
You can export an API specification or example request to use with [Postman](#using-postman) or the API platform of your choice.
```

### Request examples

As you fill in request parameters in VTEX API reference, as described above in steps 4 to 8, the request box is automattically updated. You can find this box below the `Base URL` field described above.

![request example](./making-your-first-request-4)

> This means you can copy this content to be imported into other software.

Reviewer: Acho que essa parte poderia vir no final. Só faz sentido o usuário copiar a request depois de ter exportado as especificações, nopes?

Author: Entendo como duas formas paralelas de exportar request pra trazer pro postman. Se baixou o OAS e importou no Postman, nãoprecisa copiar o exemplo de código

> ### Downloading API specifications

Reviewer: Não vejo necessidade desse título aqui caso considere a sugestão anterior
